K0s vcluster ends up in crash loop if it ever fails

### What happened?

```console
$ kubectl logs vcluster-0 
Defaulted container "syncer" out of: syncer, vcluster (init)
2023-12-07 15:33:42     INFO    setup/initialize.go:86  k0s config secret detected, syncer will ensure that it contains service CIDR    {"component": "vcluster"}
2023-12-07 15:33:42     INFO    k0s/k0s.go:58   Starting k0s    {"component": "vcluster", "args": "/k0s-binary/k0s controller --config=/etc/k0s/config.yaml --data-dir=/data/k0s --status-socket=/run/k0s/status.sock --disable-components=konnectivity-server,kube-scheduler,csr-approver,kube-proxy,coredns,network-provider,helm,metrics-server,kubelet-config"}
2023-12-07 15:33:43     INFO    setup/controller_context.go:221 couldn't retrieve virtual cluster version (Get "https://localhost:6443/version": dial tcp [::1]:6443: connect: connection refused), will retry in 1 seconds     {"component": "vcluster"}
2023-12-07 15:33:43     INFO    setup/controller_context.go:221 couldn't retrieve virtual cluster version (Get "https://localhost:6443/version": dial tcp [::1]:6443: connect: connection refused), will retry in 1 seconds     {"component": "vcluster"}
2023-12-07 15:33:44     INFO    loghelper/klog.go:24    Usage of deprecated component name      {"component": "vcluster", "component": "k0s", "time": "2023-12-07 15:33:44", "level": "warning"}
2023-12-07 15:33:44     INFO    loghelper/klog.go:24            {"component": "vcluster", "component": "k0s", "time": "2023-12-07 15:33:44", "level": "warning", "pre-flight-check": "os/NET", "property": "no kernel config found"}
2023-12-07 15:33:44     INFO    loghelper/klog.go:24    Error: failed to initialize runtime config: an instance of k0s is already running       {"component": "vcluster", "component": "k0s"}
2023-12-07 15:33:44     INFO    setup/controller_context.go:221 couldn't retrieve virtual cluster version (Get "https://localhost:6443/version": dial tcp [::1]:6443: connect: connection refused), will retry in 1 seconds     {"component": "vcluster"}
2023-12-07 15:33:44     INFO    setup/initialize.go:107 Error running k0s: exit status 1        {"component": "vcluster"}
```

You probably want to remove an existing k0s socket if one exists when starting.

### What did you expect to happen?

An error should not leave the vcluster in a crash loop

### How can we reproduce it (as minimally and precisely as possible)?

Run a 0.18.0 k0s vcluster, crash the cluster somehow

### Anything else we need to know?

_No response_

### Host cluster Kubernetes version

<details>

```console
$ kubectl version
Client Version: v1.28.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.25.11
```

</details>


### Host cluster Kubernetes distribution

<details>

```
Bare-metal plain Kubernetes install
```

</details>


### vlcuster version

<details>

```console
$ vcluster --version
vcluster version 0.18.0
```

</details>


### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
k0s
```

</details>


### OS and Arch

<details>

```
OS: Red Hat Enterprise Linux release 8.9 (Ootpa)
Arch: amd64
```
</details>

--- a/pkg/k0s/k0s.go
+++ b/pkg/k0s/k0s.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
 
 	"github.com/ghodss/yaml"
@@ -20,7 +21,16 @@ type k0sCommand struct {
 	Args    []string `json:"args,omitempty"`
 }
 
+const runDir = "/run/k0s"
+
 func StartK0S(ctx context.Context) error {
+	// make sure we delete the contents of /run/k0s
+	dirEntries, _ := os.ReadDir(runDir)
+	for _, entry := range dirEntries {
+		_ = os.RemoveAll(filepath.Join(runDir, entry.Name()))
+	}
+
+	// start k0s binary
 	reader, writer, err := os.Pipe()
 	if err != nil {
 		return err

--- a/pkg/telemetry/collect.go
+++ b/pkg/telemetry/collect.go
@@ -193,14 +193,17 @@ func (d *DefaultCollector) RecordStatus(ctx context.Context) {
 }
 
 func (d *DefaultCollector) RecordStart(ctx context.Context) {
+	chartInfo := d.getChartInfo(ctx)
 	properties := map[string]interface{}{
 		"vcluster_version":            SyncerVersion,
-		"vcluster_k8s_distro":         d.getChartInfo(ctx).Name,
 		"vcluster_k8s_distro_version": d.getVirtualClusterVersion(),
 		"host_cluster_k8s_version":    d.getHostClusterVersion(),
 		"os_arch":                     runtime.GOOS + "/" + runtime.GOARCH,
-		"helm_values":                 d.getChartInfo(ctx).Values,
 		"creation_method":             d.config.InstanceCreator,
+	}
+	if chartInfo != nil {
+		properties["vcluster_k8s_distro"] = chartInfo.Name
+		properties["helm_values"] = chartInfo.Values
 	}
 
 	// build the event and record
@@ -231,7 +234,10 @@ func (d *DefaultCollector) RecordError(ctx context.Context, severity ErrorSeveri
 
 	// if panic or fatal we add the helm values
 	if severity == PanicSeverity || severity == FatalSeverity {
-		properties["helm_values"] = d.getChartInfo(ctx).Values
+		chartInfo := d.getChartInfo(ctx)
+		if chartInfo != nil {
+			properties["helm_values"] = chartInfo.Values
+		}
 	}
 
 	// build the event and record

